dask_glm/algorithms.py

```diff
@@ -134,12 +135,28 @@ def newton(X, y, max_iter=50, tol=1e-8, family=Logistic):
     return beta
 
 
-def admm(X, y, lamduh=0.1, rho=1, over_relax=1,
+def admm(X, y, reg=L1, lamduh=0.1, rho=1, over_relax=1,
          max_iter=100, abstol=1e-4, reltol=1e-2, family=Logistic):
 
     pointwise_loss = family.pointwise_loss
     pointwise_gradient = family.pointwise_gradient
 
+    def create_local_gradient(func):
+        @functools.wraps(func)
+        def wrapped(beta, X, y, z, u, rho):
+            return func(beta, X, y) + rho * (beta - z + u)
+        return wrapped
+
+    def create_local_f(func):
```

Comment:
Could this be renamed perhaps? The name doesnt tell me what this function is trying to do. Feel free to ignore if this doesn't make sense at this point.